Tests/UnitTests/FileShareClientTests/DownloadFilesTests.cs

```diff
@@ -126,5 +127,116 @@ public async Task TestDownloadFileSetsAuthorizationHeader()
+            string destDownldLocation = Path.Combine(Environment.CurrentDirectory, "AFilename.txt");
+            nextResponseStatusCode = HttpStatusCode.BadRequest;
+            var destStream = new FileStream(destDownldLocation, FileMode.OpenOrCreate, FileAccess.Write, FileShare.ReadWrite);
+           
+            var result = await fileShareApiClient.DownloadFileAsync(batchId, "AFilename.txt", destStream, 10, CancellationToken.None);
+            
+            Assert.AreEqual((int)nextResponseStatusCode, result.StatusCode);
+            Assert.AreEqual($"/basePath/batch/{batchId}/files/AFilename.txt", lastRequestUri.AbsolutePath);
+        }
+
+
+        [Test]
+        public async Task TestDownloadFilesForABatchWithAFileThatDoesNotExistWithCancellationToken()
+        {
+            var batchId = Guid.NewGuid().ToString();
+            nextResponseStatusCode = HttpStatusCode.NotFound;
+            string destDownldLocation = Path.Combine(Environment.CurrentDirectory, "AFilename.txt");
+            var destStream = new FileStream(destDownldLocation, FileMode.OpenOrCreate, FileAccess.Write, FileShare.ReadWrite);
+            
+            var result = await fileShareApiClient.DownloadFileAsync(batchId, "AFilename.txt", destStream, 10, CancellationToken.None);
+            
+            Assert.AreEqual((int)nextResponseStatusCode, result.StatusCode);
+            Assert.AreEqual($"/basePath/batch/{batchId}/files/AFilename.txt", lastRequestUri.AbsolutePath);
+        }
+
+        [Test]
+        public async Task TestGetBatchStatusForABatchThatHasBeenDeletedWithCancellationToken()
+        {
+            var batchId = Guid.NewGuid().ToString();
+            string destDownldLocation = Path.Combine(Environment.CurrentDirectory, "AFilename.txt");
+            nextResponseStatusCode = HttpStatusCode.Gone;
+            var destStream = new FileStream(destDownldLocation, FileMode.OpenOrCreate, FileAccess.Write, FileShare.ReadWrite);
+            
+            var result = await fileShareApiClient.DownloadFileAsync(batchId, "AFilename.txt", destStream, 10, CancellationToken.None);
+            
+            Assert.AreEqual((int)nextResponseStatusCode, result.StatusCode);
+            Assert.AreEqual($"/basePath/batch/{batchId}/files/AFilename.txt", lastRequestUri.AbsolutePath);
+        }
+
+        [Test]
+        public async Task TestForDownloadFilesWhenFileSizeIsGreaterThanMaxDownlodBytes()
+        {
+            var batchId = Guid.NewGuid().ToString();
+            string destDownldLocation = Path.Combine(Environment.CurrentDirectory, "AFilename.txt");
+            nextResponseStatusCode = HttpStatusCode.PartialContent;
+            var destStream = new FileStream(destDownldLocation, FileMode.OpenOrCreate, FileAccess.Write, FileShare.ReadWrite);
+            nextResponse = new MemoryStream(10585760);
+
+            var result = await fileShareApiClient.DownloadFileAsync(batchId, "AFilename.txt", destStream, 10585760, CancellationToken.None);
```

Comment:
When debugging this test it shows the range for the first chunk is 0 to 10485760 (10485761 bytes), which exceeds the `maxDownloadBytes` setting of 10485760. Could you please fix `DownloadFileAsync` to correctly honour the `maxDownloadBytes` setting.

I would also like to see this test changed to verify the correct ranges are being used for each download request

Reply:
changes done for implementation. Thanks